src/NuGetPackageVerifier/Rules/AssemblyHasCommitHashAttributeRule.cs

```diff
@@ -0,0 +1,39 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using Mono.Cecil;
+using Mono.Collections.Generic;
+
+namespace NuGetPackageVerifier.Rules
+{
+    public class AssemblyHasCommitHashAttributeRule : AssemblyHasAttributeRuleBase
+    {
+        public override IEnumerable<PackageVerifierIssue> ValidateAttribute(
+            string currentFilePath,
+            Collection<CustomAttribute> assemblyAttributes)
+        {
+            var fileName = Path.GetFileNameWithoutExtension(currentFilePath);
+            var endsWithSources = fileName.EndsWith(".Sources", System.StringComparison.InvariantCulture);
```

Comment:
100% never use StringComparison.InvariantCulture - use StringComparison.Ordinal - or perhaps StringComparison.OrdinalIgnoreCase because that should be fine too.

On the nitty side, this variable might be better called `isSourcesPackage` because that's its semantic purpose.